.travis.yml

```diff
@@ -1,28 +1,65 @@
+# since we don't need sudo, travis recommend xenial
+dist: xenial
+
 language: node_js
 
-os: linux
+# we don't need previous commits, so don't need default depth of 50
+git:
+  depth: 1
 
-matrix:
-  include:
-  - node_js: '8'
-  - node_js: '10'
+# have to enable branch build to trigger on git tag but
+# we don't want the same job to run twice on pull request
+branches:
+  only:
+    - master
+    - dev
+    - /^v\d+\.\d+\.\d*$/ # Ensure to build release tags
 
-sudo: false
+stages:
+  - test
+  - name: publish
+    if: tag =~ ^v\d+\.\d+\.\d*$
 
 install:
-- npm install
-script:
-  - npm run test:ci
+  - npm install
 
-after_success:
-- cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage
-
-deploy:
-  provider: npm
-  email: $NPM_EMAIL_ACCOUNT
-  api_key:
-    secure: Iu5Q2uuiF+jRWkNRtX77YElMwyEopzKD+iHQQkT7HheH12AwDAW/PZiwLRhBTif8kvHmIIwPhjey6i7QwmJFUmpb433299WF+tinRhy5mBFntdaNmgS2Wvv/hOFJza2XdOeIsKA+s+zPAGArfLonmvQX5QKVmJI7/W+hyqJT0FjtTMqxPyJ00ENuDeZdUEcsaoDAwu1XTA6bj5ILnguR+smeGoZscpp8nwTv27fA+GWLFGf08pPAyuHSwc1/lRxSbLlw/f0ZqnCclGU+SF2EV4/9+xxcPWqtJdJfURR/LaOkg3Q5o4aILRjNJNCbpRkrc5lEcPEvBPLrI6e7F9ssg08aHniaMvdD5B9TsgGqV6sxZX/KvMIHWfzBRLuCzY2bPE8Wbw+ZI/yi9+89NZdRikAMyWocvg4tx1OsQRcrRltsXqQrZyw45IR5oSv0kLx54xrslXu8jjP08jQEpnBncJp883DX6DmO42TRP4QYHsQ8BB5hHQ2qmVoKMydkLy0Kdvq7e+9v/pZHEMsR/HOwz1S5xYsSeqwqoq4u/H+hPFWu3GVJXbsVLHZuaDU3fBRMtJOcjujPszrvN/cNt2JryTT1EYPJotmpURqu/n9vPLMzoi1wIKPrZiAcEzh8MLlM93MyG/yuPeHZOvMuwCVtOXd3KKH8sEJvbTVUtKIvQXw=
-  on:
-    tags: true
-    repo: serverless/serverless-azure-functions
-    branch: master
+jobs:
+  include:
+    - stage: test
+      name: "Unit Tests"
+      node_js: 
+        - "8"
+        - "10"
+      script:
+        - npm run test:ci
+      after_success:
+        - cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage
+    
+    - stage: publish
+      name: "NPM release"
+      node_js: "10"
+      before_deploy: 
+        - npm run build
+      deploy:
+        - provider: npm
+          email: $NPM_EMAIL_ACCOUNT
+          api_key:
+            secure: Iu5Q2uuiF+jRWkNRtX77YElMwyEopzKD+iHQQkT7HheH12AwDAW/PZiwLRhBTif8kvHmIIwPhjey6i7QwmJFUmpb433299WF+tinRhy5mBFntdaNmgS2Wvv/hOFJza2XdOeIsKA+s+zPAGArfLonmvQX5QKVmJI7/W+hyqJT0FjtTMqxPyJ00ENuDeZdUEcsaoDAwu1XTA6bj5ILnguR+smeGoZscpp8nwTv27fA+GWLFGf08pPAyuHSwc1/lRxSbLlw/f0ZqnCclGU+SF2EV4/9+xxcPWqtJdJfURR/LaOkg3Q5o4aILRjNJNCbpRkrc5lEcPEvBPLrI6e7F9ssg08aHniaMvdD5B9TsgGqV6sxZX/KvMIHWfzBRLuCzY2bPE8Wbw+ZI/yi9+89NZdRikAMyWocvg4tx1OsQRcrRltsXqQrZyw45IR5oSv0kLx54xrslXu8jjP08jQEpnBncJp883DX6DmO42TRP4QYHsQ8BB5hHQ2qmVoKMydkLy0Kdvq7e+9v/pZHEMsR/HOwz1S5xYsSeqwqoq4u/H+hPFWu3GVJXbsVLHZuaDU3fBRMtJOcjujPszrvN/cNt2JryTT1EYPJotmpURqu/n9vPLMzoi1wIKPrZiAcEzh8MLlM93MyG/yuPeHZOvMuwCVtOXd3KKH8sEJvbTVUtKIvQXw=
```

Comment:
just double checking, but these are the encrypted values from running the travis tool, right? :slightly_smiling_face:

Reply:
Yup 👍